cmd/bolt/main.go

```diff
@@ -416,9 +419,137 @@ func (cmd *DumpCommand) PrintPage(w io.Writer, r io.ReaderAt, pageID int, pageSi
 // Usage returns the help message.
 func (cmd *DumpCommand) Usage() string {
 	return strings.TrimLeft(`
-usage: bolt dump -page PAGEID PATH
+usage: bolt dump PATH pageid [pageid...]
 
-Dump prints a hexadecimal dump of a single page.
+Dump prints a hexadecimal dump of one or more pages.
+`, "\n")
+}
+
+// DumpItemCommand represents the "dump-item" command execution.
+type DumpItemCommand struct {
+	Stdin  io.Reader
+	Stdout io.Writer
+	Stderr io.Writer
+}
+
+// newDumpItemCommand returns a DumpItemCommand.
+func newDumpItemCommand(m *Main) *DumpItemCommand {
+	return &DumpItemCommand{
+		Stdin:  m.Stdin,
+		Stdout: m.Stdout,
+		Stderr: m.Stderr,
+	}
+}
+
+type dumpItemOptions struct {
+	help  bool
+	key   bool
+	value bool
+}
+
+// Run executes the command.
+func (cmd *DumpItemCommand) Run(args ...string) error {
+	// Parse flags.
+	options := &dumpItemOptions{}
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+	fs.BoolVar(&options.key, "key", false, "Print out the raw bytes of the leaf item key.")
+	fs.BoolVar(&options.value, "value", false, "Print out the raw bytes of the leaf item value.")
+	fs.BoolVar(&options.help, "h", false, "")
+	if err := fs.Parse(args); err != nil {
+		return err
+	} else if options.help {
+		fmt.Fprintln(cmd.Stderr, cmd.Usage())
+		return ErrUsage
+	}
+
+	if options.key && options.value {
```

Comment:
why not allow both to be set?

Reply:
That sounds reasonable. Maybe we should do something more like etcd where by default we print key and value, and provide for `--key-only` and `--value-only` flags.

Reply:
right. i hope user can use the command without specifying any flag by default. right now they have to say --key or --value. If no option is provided, it will fail. not a good UX in my opinion.